client.go

```diff
@@ -590,6 +590,8 @@ func (c *Client) checkResponse(resp *http.Response, err error) (*http.Response,
 func (c *Client) getErrorFromResponse(resp *http.Response) APIError {
 	// check whether the error response is declared as JSON
 	if !strings.HasPrefix(resp.Header.Get("Content-Type"), "application/json") {
+		defer func() { _ = resp.Body.Close() }()
```

Comment:
nit: i suppose this is the same as defer resp.Body.Close() if we don't need to check the error